packages/@uppy/aws-s3-multipart/types/index.d.ts

```diff
@@ -33,29 +34,29 @@ export interface AwsS3MultipartOptions extends PluginOptions {
     ) => MaybePromise<{ uploadId: string; key: string }>
     listParts?: (
       file: UppyFile,
-      opts: { uploadId: string; key: string; signal: AbortSignal }
+      opts: { uploadId: string; key: string }
     ) => MaybePromise<AwsS3Part[]>
-    getTemporarySecurityCredentials?: boolean | (() => MaybePromise<AwsS3STSResponse>)
     signPart?: (
       file: UppyFile,
-      opts: { uploadId: string; key: string; partNumber: number; body: Blob, signal: AbortSignal }
+      opts: { uploadId: string; key: string; partNumber: number; body: Blob; signal: AbortSignal }
     ) => MaybePromise<AwsS3SignedPart>
     /** @deprecated Use signPart instead */
     prepareUploadParts?: (
       file: UppyFile,
-      partData: { uploadId: string; key: string; parts: [{ number: number, chunk: Blob }], signal: AbortSignal }
-    ) => MaybePromise<{ presignedUrls: { [k: number]: string }, headers?: { [k: string]: string } }>
+      partData: { uploadId: string; key: string; parts: [{ number: number, chunk: Blob }] }
+    ) => MaybePromise<{ presignedUrls: { [k: number]: string }, headers?: Record<number, Record<string, string>> }>
```

Comment:
The webpage documentation (and examples) do not match the types. In particular it looks like the types are missing an object level.

From the docs as an example return value:
```
{
    "presignedUrls": {
        "1": "https://bucket.region.amazonaws.com/path/to/file.jpg?partNumber=1&...",
        "2": "https://bucket.region.amazonaws.com/path/to/file.jpg?partNumber=2&...",
        "3": "https://bucket.region.amazonaws.com/path/to/file.jpg?partNumber=3&..."
    },
    "headers": {
        "1": { "Content-MD5": "foo" },
        "2": { "Content-MD5": "bar" },
        "3": { "Content-MD5": "baz" }
    }
}
```

Reply:
Why do we have `k: number` for `presignedUrls` and `Record` for `headers`?

Reply:
I don't know if there is any sort of Uppy standard as to the preferred approach. `Record` is definitely used throughout the codebase as well as the {[k: type]: type} syntax. For example the `core` types use that style `packages/@uppy/core/types/index.d.ts`. I think I would suspect that the `core` style to be more likely to be the preferred approach.

Either way styles should probably not be mixed across a single file, or worse a single function type declaration. Thus I have updated the types here to all use Record.